### PR TITLE
Fix reporting issues found by Kibit

### PR DESCRIPTION
**Pull request by lukaszkorecki**
Hi! 👋 

I got everything working again and here are results:

**running `codeclimate analyze -e kibit` with a container build based on current master branch**

**running `codeclimate analyze -e kibit` with a container build based on my fork**

In both instances, it's checking a sample project which can be found in `dev-resources`. 
I've also split the code into the CLI/runner part and the reporter itself so that it's easier to use them from the REPL.

## TODO:

- [x] make sure that final file list for analysis is unique 
- [x] add unit tests 
- [x] make sure that `exclude_paths` option is honored 
- [x] produce smaller Docker image  (it's working already, but it's going to be a follow-up PR as it's not related to the changes here)
- [x] clean up namespace declarations

### CODE REVIEW

File: src/codeclimate/reporter.clj [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+(defn target-files
+  [config]
+  (->> (:include_paths config)
+       (map #(cond
+               (string? %) (io/file %)
+               :else %))
+       (map kibit.driver/find-clojure-sources-in-dir)
+       flatten
+       distinct))
```
Comment (lukaszkorecki) [inline src/codeclimate/reporter.clj]: `distinct` doesn't work here because we're dealing with a lazyseq of File objects.

**Comment by lukaszkorecki** [comment]
🙌 it also now catches an issue in this very project:

**Comment by dblandin** [comment]
@lukaszkorecki Thanks for fixing this!

I've gone ahead and updated the plugin in production with an image built from https://github.com/lukaszkorecki/codeclimate-kibit/commit/28876258896e08c3d237f64df4788c0abdf66e79.

I'm still hoping to see this pull request merged but in the meantime, we can keep iterating on your fork.

:tada: :100: :clap:

**Comment by lukaszkorecki** [comment]
Hi @dblandin 👋 Great! Can't wait to see it working in our team repos 🙌 

@andrewhr - I'd be more than happy to take over the maintenance of this plugin if you don't have the time 👍

**Comment by lukaszkorecki** [comment]
@dblandin I just pushed: https://github.com/lukaszkorecki/codeclimate-kibit/commit/14480f1538303bf2f9ce4d5a8882d07c5da2d8d7 - it addresses the issue when kibit sometimes doesn't report the line number (I can reproduce it so it will have to be a fix in Kibit itself)

**Comment by andrewhr** [comment]
@lukaszkorecki first and foremost, thank you a lot for all those contributions!

As I'm not using it by myself, I get really outdated. I'm sorry for taking so long to look into it, but I want to make you sure I really appreciate all the work you put in here.

Now that the spec has settled and given kibit is very stable (interface wise) I don't imagine there is a need to much work to keep it updated. But, given context, I'm ok with handling the maintenance of this project over you. If that's the direction we want to make, @dblandin can you help us?

**Comment by lukaszkorecki** [comment]
@andrewhr No worries - completely understand the time constraints!

And as I wrote before - super happy to take over the maintenance 👍 cc @dblandin 
There's few things I'd like to improve and there's at least one bug in Kibit itself which needs addressing (if there's an issue within `#( ... )` macro the location is null)

**Comment by dblandin** [comment]
@andrewhr @lukaszkorecki 

There are a few paths forward regarding maintenance:

1. Keep the plugin here at andrewhr/codeclimate-kibit and give @lukaszkorecki push access.
1. Update the README to refer to @lukaszkorecki's fork and let that be the maintained version going forward.
1. Move the repository under the @codeclimate-community organization and I can grant push access.

I don't have any particular preference. Totally up to you two!

Thanks! :+1:

**Comment by lukaszkorecki** [comment]
@dblandin @andrewhr Feels like **3** is the most future-proof and the most "official". (I'm fine with any of the proposed paths)

**Comment by davehenton** [comment]
Hey @lukaszkorecki, Dave here with Code Climate. I’d love to help get this repo moved over to our @codeclimate-community organization, with you as a maintainer, so we can share the maintenance load. Just let us know. Thanks!

**Comment by lukaszkorecki** [comment]
@davehenton Sounds good, let's do it :thumbsup:

**Comment by davehenton** [comment]
@lukaszkorecki thanks! Feel free to transfer following these steps: https://help.github.com/en/articles/transferring-a-repository

**Comment by lukaszkorecki** [comment]
@davehenton this got a bit complicated as my fork is based on `codeclimate/codeclimate-kibit`: https://github.com/codeclimate/codeclimate-kibit/network/members which means I cannot transfer the ownership as the repository exists. 
I think best way to deal with this situation would be to:

- merge changes from my fork into `codeclimate/codeclimate-kibit`
- then grant me commit access so I can carry on the maintenance

(I'm also slowly working on an engine for https://github.com/borkdude/clj-kondo which would also be great to have in Codeclimate)

**Comment by davehenton** [comment]
Hey @lukaszkorecki, thanks for your patience with this. 

We would actually like to transfer the ownership to our community org (https://github.com/codeclimate-community). We will give then give you committer access, and this way we can help him merge anything if needed.

Let me know if that works for you, or if you need any help getting that implemented. I'll be standing by.

**Comment by lukaszkorecki** [comment]
@davehenton no problem, delays are absolutely understandable.

I was thinking of combining multiple linters into one tool and also potentially adding a tool which tracks cyclomatic complexity of Clojure code (no promises on that one yet!) - so `codeclimate-kibit` name wouldn't really fit here. What would be the advisory? Have multiple engines? Or have one, called `codeclimate-clojure` ?

**Comment by maxjacobson** [comment]
@lukaszkorecki Hi Łukasz, Max from CC here. One important goal here is to maintain continuity of service for existing users of the kibit engine. To that end, I think it would be best to continue to offer this engine, and add additional ones for non-kibit tools. WDYT?

**Comment by lukaszkorecki** [comment]
@maxjacobson sounds good! I'll transfer the repo to codeclimate-community later today

**Comment by davehenton** [comment]
@lukaszkorecki I suspect you ran into an issue transferring the repo to `codeclimate-community` due to permissions. We'd like to fork your repo and add it to `codeclimate-community`, then grant you access. Does that sound like a plan?

**Comment by lukaszkorecki** [comment]
@davehenton Yeah, it appears I need to have permissions to create new repository in that org. I think forking https://github.com/lukaszkorecki/codeclimate-kibit/ will be the best approach 👍

**Pull request merged**